src/components/PopupWithForm.js

```diff
@@ -22,8 +24,13 @@ export class PopupWithForm extends Popup {
     setEventListeners() {
         this.form.addEventListener("submit", (event) => {
             event.preventDefault();
+            this._submitButton.textContent = "Saving...";
             const inputValues = this._getInputValues();
-            this._submitHandler(event, inputValues);
+            this._submitHandler(event, inputValues)
+                .then(() => { 
+                    console.log(this._buttonText);
+                    this._submitButton.textContent = this._buttonText;
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.